app/controllers/v0/yellow_ribbon_programs_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module V0
+  class YellowRibbonProgramsController < ApiController
+    include Facets
+
+    # GET /v0/yellow_ribbon_programs
+    # ?page=1
+    # &per_page=30
+    # &sort_by=number_of_students
+    # &sort_direction=desc
+    # &city=boulder
+    # &country=usa
+    # &contribution_amount=unlimited
+    # &number_of_students=unlimited
+    # &school_name_in_yr_database=university
+    # &state=co
+    def index
+      @meta = {
+        version: @version,
+        count: search_results.count
+      }
+      render json: search_results.page(params[:page]), meta: @meta
+    end
+
+    private
+
+    def normalized_query_params
+      query = params.deep_dup
+      query.tap do
+        # Filtering query params.
+        query[:city].try(:strip!)
+        query[:city].try(:downcase!)
+
+        query[:country].try(:strip!)
+        query[:country].try(:downcase!)
+
+        query[:contribution_amount].try(:strip!)
+        query[:contribution_amount].try(:downcase!)
+
+        query[:number_of_students].try(:strip!)
+        query[:number_of_students].try(:downcase!)
+
+        query[:school_name_in_yr_database].try(:strip!)
+        query[:school_name_in_yr_database].try(:downcase!)
+
+        query[:state].try(:strip!)
+        query[:state].try(:downcase!)
+
+        # Sorting query params.
+        query[:sort_by].try(:strip!)
+        query[:sort_by].try(:downcase!)
+
+        query[:sort_direction].try(:strip!)
+        query[:sort_direction].try(:downcase!)
+      end
+    end
+
+    def search_results
+      @query ||= normalized_query_params
+
+      # Derive the search results.
+      results = YellowRibbonProgram.version(@version).search(@query)
```

Comment:
`@version` is a `Version` object.  The `YellowRibbonProgram#version` attribute is an integer.
So, this should be
```suggestion
      results = YellowRibbonProgram.version(@version.number).search(@query)
```
Your specs are passing because when you run them `@version.id` and `@version.number` happen to be the same integer, and Rails uses the `id` in the SQL it generates.

Reply:
So I think we want to use `@version.id`, not `@version.number`, as per one of your last comments, right? https://github.com/department-of-veterans-affairs/gibct-data-service/pull/584#discussion_r387851540

I made the update to use `version.id` explicitly in the model! 🙂

Reply:
Updated to use `version.number` due to:

![image](https://user-images.githubusercontent.com/12773166/76006177-583c3680-5ec9-11ea-8f46-e89985fb7365.png)